### DISCUSSION TITLE
[Feature Request] Allow Custom Sorting Specs inside plugin settings

### DISCUSSION THREAD

**Discussion by windblower420** [SebastianMC/obsidian-custom-sort#34] [Ideas]
Would be great for the plugin settings to allow the user to define the sorting options instead of needing a note. I wanted to at least hide the note using ".sortspec.md" but it won't read it if its hidden. 

a plugin that allows this is called supercharged links https://github.com/mdelobelle/obsidian_supercharged_links

maybe the settings this plugin provides will inspire and make it easier to provide

This is a great plugin that a lot of people will love! thanks for your work

**Comment by SebastianMC** [comment]
Hi @Obadiahcruz !

when creating the plugin, the approach was to _keep the sorting rules specification intentionally in a note_ (or spread over more notes). This is to be able to copy the sorting when copying (or moving) the notes to a different folder or a different vault or when archiving the notes.
For that purpose, the sorting specification can reside in:
- sortspec.md note(s), and in
- the folder note (a note named exactly as the parent folder), and in
- the designated note anywhere in the vault (path to the note is set in plugin settings).

Keeping the yaml in front matter of some note has the additional advantage of obsidian notes editor support - basic syntax highlighting, collapsing the `sorting-spec: |` value, etc. 

The plugin also supports an experimental undocumented feature which allows hiding of the `sortspec.md` note (or any other notes or folders) in file explorer via the syntax:

```
/--hide: sortspec.md
```

Yet, this feature is experimental and intentionally kept undocumented because it could easily confuse the user:
- if a note or folder is hidden, the function `Reveal file in navigation` will obviously not work if the custom sorting is active
- if a folder is not included (or explicitly excluded) from custom sorting, the hiding syntax will be ignored for their files or folders.

Allowing the user to specify sorting configuration in a text field of plugin settings is an easy task, from development perspective. Yet because of the reasons listed above I intentionally resigned from adding it.

Obviously no problem with adding it if people find it useful

Before that, could you please see maybe the below approach works for you?

If you put the below sorting specification in the file `sortspec.md` in root folder, it will:
- let the plugin take over the control of sorting of the root `/` folder and mimic the default Obsidian sorting (folders go first, then go files, all sorted alphabetically)
- let the `sortspec.md` file be hidden in File Explorer 

```
---
sorting-spec: |
 // Let the plugin take over the control of sorting in the root folder and mimic default Obsidian sorting
 target-folder: /
 /folders ...
  < a-z
 /:files
  < a-z
 
 /--hide: sortspec.md
---
```
You can extend the above specification with sorting rules for other (sub)folders, as needed

Let me know what you think

**Comment by SebastianMC** [comment]
Hi @Obadiahcruz!
not having any further feedback from you, I'm about to convert this ticket into a discussion.
Seeing how people struggle with the basic step of putting a basic `sortspec.md` to work I'm looking towards some less technical and more user-friendly options. While introducing the 'sorting specification in plugin setting page' is contrary to some basic ideas behind the plugin and it poses its own challenges, still it looks like one of viable long-term directions for this plugin.

Note: if anyone is reading this comment here, you are already viewing it in **[Discussions](https://github.com/SebastianMC/obsidian-custom-sort/discussions/34)** section of the plugin home in GitHub

**Comment by Genei180** [comment]
Hey, Just wanted to let you know, that I was just looking for Alphabetical Sorting including Folders.
This Plugin may be completely Overkill for this task, but the fact that Obsidian doesn't do that Natively is driving me nuts.

I will definitely use the `/--hide: sortspec.md` feature.
Originally I would have looked to make the Setting Just Universal and store the File Alongside the plugins "custom-sort" Folder hidden away, and the Vault pretty clean.

So thank you good Sir for making this quite thought true Plugin.
